### DISCUSSION TITLE
Unexpected EOF

### DISCUSSION THREAD

**Discussion by Caden64** [HypixelDev/PublicAPI#522] [Q&A] [answered]
I have go code that pretty quickly gets the skyblock auctions API using wait groups that when run returns with Unexpected EOF as the response from the function basic data found on line 52 to 60 in utils.go 

specifically the error: `Error with request: Get "https://api.hypixel.net/skyblock/auctions": unexpected EOF`
`Nothing returned`
`error doing request`
`Program failed: error doing request`

when I run the code in main.go as an infinite loop without wait groups it runs successfully for hours at a time

hypixel code help: https://hypixel.net/threads/unexpected-eof-when-requesting-auction-house.4857311/#post-35023603

Repository: https://github.com/YoHypixel/AuctionParcer

**Comment by Caden64** [comment, marked as answer]
Added a 200 millisecond delay

**Comment by GrizzlT** [comment]
I don't know for sure, but maybe it has to do with spawning 200 concurrent `HttpClient`s?

I'm not an expert at network code but I heard that reusing the same "`connection pool`" across threads could improve your performance (and probably avoid that error). The delay that fixed it might just give every client enough time to receive a response, it certainly wouldn't feel like a "permanent" or "perfect" solution to me (though we all know how quick fixes are oh so convenient).

Though once again, I'm not an expert at http clients nor go-lang (more of a rust dev [my 2nd crate ever!](https://crates.io/crates/hypixel_api)) but maybe some of the problems I encountered carry over? If I'm completely wrong, don't hesitate to tell me too :)
